graph/quadwriter.go

```diff
@@ -59,13 +69,50 @@ func (h *Handle) Close() {
 }
 
 var (
-	ErrQuadExists   = errors.New("quad exists")
-	ErrQuadNotExist = errors.New("quad does not exist")
+	ErrQuadExists    = errors.New("quad exists")
+	ErrQuadNotExist  = errors.New("quad does not exist")
+	ErrInvalidAction = errors.New("invalid action")
 )
 
+// DeltaError records an error and the delta that caused it.
+type DeltaError struct {
+	Delta Delta
+	Err   error
+}
+
+func (e *DeltaError) Error() string {
+	return e.Delta.Action.String() + " " + e.Delta.Quad.String() + ": " + e.Err.Error()
+}
+
+// IsQuadExist returns whether an error is a DeltaError
+// with the Err field equalt to ErrQuadExists.
+func IsQuadExist(err error) bool {
+	de, ok := err.(*DeltaError)
+	return ok && de == ErrQuadExists
+}
+
+// IsQuadExist returns whether an error is a DeltaError
+// with the Err field equalt to ErrQuadNotExist.
+func IsQuadNotExist(err error) bool {
+	de, ok := err.(*DeltaError)
+	return ok && de == ErrQuadNotExist
+}
+
+// IsQuadExist returns whether an error is a DeltaError
+// with the Err field equalt to ErrInvalidAction.
+func IsInvalidAction(err error) bool {
```

Comment:
Might not be necessary. But it's ok to have it if you think it will be useful widely.